enterprise/internal/a8n/service.go

```diff
@@ -788,6 +814,16 @@ func (s *Service) UpdateCampaign(ctx context.Context, args UpdateCampaignArgs) (
 	}
 	partiallyPublished := !published && status.Total != 0
 
+	if campaign.CampaignPlanID != 0 && args.Branch != nil {
+		if published {
```

Comment:
We need to make sure that we also don't allow updating the branch if the Campaign has been partially published:

```suggestion
		if published || partiallyPublished {
```